.pre-commit-config.yaml

```diff
@@ -0,0 +1,43 @@
+repos:
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.238
+    hooks:
+      - id: ruff
+        args: [--select, "B,C4,E,F,W,RUF,I"]
+        exclude: ^resources
+  - repo: https://github.com/abravalheri/validate-pyproject
+    rev: v0.12.1
+    hooks:
+      - id: validate-pyproject
+
+  - repo: https://github.com/psf/black
+    rev: 22.12.0
+    hooks:
+      - id: black
+        args: [--check]
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.991
+    hooks:
+    -   id: mypy
+        args: [ --ignore-missing-imports]
```

Comment:
I think it may be wiser to use a config file for the template or have the pre-commit hook called directly, otherwise the configs may start drifting apart.

Reply:
Indeed, I also don't like it, but pre-commit does not support toml. When I looked into how did smarter people than me have it set up, the repos I was checking also had the configs separate.  
I can see the reasoning for doing that, but it's definitely not ideal.

Are you suggesting to have all the linting in pre-commit yaml and run ’pre-commit --all-files' during in CI?

Reply:
I don't really know the `pre-commit` tool; but isn't there a `recipe` for just calling the `./hooks/precommit` script? This would alleviate the need to maintain the yaml further

Reply:
If you mean the github action, that also uses the yaml.  There is also https://pre-commit.ci/

Reply:
Yeah so I'm unfamiliar with all of these tools. I assumed `.pre-commit-config.yaml` was for `https://pre-commit.com/`. I'm suggesting just calling the script (maybe through `https://pre-commit.com/`) to avoid maintaining two different set of rules.

Reply:
Turns out I'm stupid and when pre-commit is running the linters, they do use the pyproject.toml arguments. 

Now, the remaining problem is how to keep the tool versions in sync, if you want to have them as package optional-dependencies (netifaces[dev])  
pre-commit has a feature to auto-update the revs to latest version - `pre-commit autoupdate` will bump the versions in yaml. But I don't think there is any good way to bump them also in pyproject.toml.  Looks like not many project do specify linters as optional dependency I guess.

Reply:
>  Looks like not many project do specify linters as optional dependency I guess.

I typically use poetry, with dev dependencies, but that conflicted with maturin. I think your solution is fine for now tbh